src/Transports/MassTransit.SqlTransport.SqlServer/Configuration/SqlServerDbTransportConfigurationExtensions.cs

```diff
@@ -8,16 +9,24 @@ namespace MassTransit
     public static class SqlServerDbTransportConfigurationExtensions
     {
         public static IServiceCollection AddSqlServerMigrationHostedService(this IServiceCollection services, bool create = true, bool delete = false)
+        {
+            services.AddSqlServerMigrationHostedService(options =>
+            {
+                options.CreateDatabase = create;
+                options.DeleteDatabase = delete;
+                options.CreateInfrastructure = create;
+            });
+
+            return services;
+        }
+
+        public static IServiceCollection AddSqlServerMigrationHostedService(this IServiceCollection services, Action<SqlTransportMigrationOptions> options)
         {
             services.AddTransient<ISqlTransportDatabaseMigrator, SqlServerDatabaseMigrator>();
 
             services.AddOptions<SqlTransportOptions>();
             services.AddOptions<SqlTransportMigrationOptions>()
-                .Configure(options =>
-                {
-                    options.CreateDatabase = create;
```

Comment:
Only minor change here, let's mirror the default options of create = true, before calling the callback. Also, can you rename the options parmaeter to 'callback' to avoid confusion. That should do it!

Reply:
FWIW, I typically do this type of cleanup myself, but if you do, I can squash/merge the PR and it shows as merged vs. closed if I do it locally from the command line.